app/internal/weekly_tasks.py

```diff
@@ -0,0 +1,138 @@
+from datetime import date, datetime, time
+from typing import Tuple
+
+from app.database.models import User, Task, WeeklyTask
+from sqlalchemy.orm.session import Session
+
+
+def check_inputs(days: str, the_time: time, title: str) -> bool:
+    """Checks inputs, used by the make_or_change_weekly_task function"""
+    if not days or days == "" or not the_time or not title or title == "":
+        return False
+    return True
+
+
+def make_or_change_weekly_task(
+    user: User, session: Session,
+    mode: str, weekly_task_id: int,
+    title: str, days: str,
+    content: str, is_important: bool, the_time: time
+) -> Tuple[bool, WeeklyTask]:
+    """This function is being used to add a Weekly Task to the user
+        or to edit an existing Weekly Task the user have.
+
+    Args:
+        user (User): The user who wants to add or edit the Weekly Task.
+        session (Session): The session to redirect to the database.
+        mode (str): Determines whether in Add or Edit mode.
+        weekly_task_id (int): In edit mode, represents the
+            weekly task being edited.
+        title (str): Title of the Weekly Task.
+        days (str): Return days of the Weekly Task.
+        content (str): Content of the Weekly Task.
+        is_important (bool): If the task is important.
+        the_time (time): Return time of the Weekly Task.
+
+    Returns:
+        Tuple: Boolean variable which shows if the change has been
+            made to the db, and the model WeeklyTask which the
+            function made so far.
+    """
+    weekly_task = WeeklyTask(
+        title=title,
+        content=content,
+        is_important=is_important,
+        owner_id=user.id
+    )
+
+    if weekly_task_id:
+        weekly_task.id = weekly_task_id
+
+    inputs_ok = check_inputs(days, the_time, title)
+    if not inputs_ok:
+        return False, weekly_task
```

Comment:
Prefer not to use [StatusCode, Result] - this behavior is not Pythonic.

Instead, prefer to raise an error if something bad happen and catch it in the caller.